apps/OpenSpace/ext/launcher/src/profile/actiondialog.cpp

```diff
@@ -0,0 +1,612 @@
+                "well?",
+                action->identifier, ghoul::to_string(kb.key)
+            )),
+            QMessageBox::StandardButton::Yes,
+            QMessageBox::StandardButton::No
+        );
+        if (button == QMessageBox::StandardButton::Yes) {
+            _keybindings.data.erase(_keybindings.data.begin() + i);
+            delete _keybindings.list->takeItem(static_cast<int>(i));
+            i--;
+        }
+        else {
+            // If the user chooses 'No' at least once, we have to bail
+            return;
+        }
+    }
+
+    for (size_t i = 0; i < _actions.data.size(); ++i) {
+        if (_actions.data[i].identifier == action->identifier) {
+            _actions.data.erase(_actions.data.begin() + i);
+            delete _actions.list->takeItem(static_cast<int>(i));
+            clearActionFields();
+            return;
+        }
+    }
+
+    ghoul_assert(false, "We shouldn't be able to get here");
+}
+
+void ActionDialog::actionSelected() {
+    const Profile::Action* action = selectedAction();
+    if (action) {
+        // Action selected
+        _actions.identifier->setText(QString::fromStdString(action->identifier));
+        _actions.identifier->setEnabled(true);
+        _actions.name->setText(QString::fromStdString(action->name));
+        _actions.name->setEnabled(true);
+        _actions.guiPath->setText(QString::fromStdString(action->guiPath));
+        _actions.guiPath->setEnabled(true);
+        _actions.documentation->setText(QString::fromStdString(action->documentation));
+        _actions.documentation->setEnabled(true);
+        _actions.isLocal->setChecked(action->isLocal);
+        _actions.isLocal->setEnabled(true);
+        _actions.script->setText(QString::fromStdString(action->script));
+        _actions.script->setEnabled(true);
+        _actions.addButton->setEnabled(false);
+        _actions.removeButton->setEnabled(true);
+        _actions.saveButtons->setEnabled(true);
+    }
+    else {
+        // No action selected
+        _actions.addButton->setEnabled(true);
+        _actions.removeButton->setEnabled(false);
+        _actions.saveButtons->setEnabled(false);
+    }
+}
+
+void ActionDialog::actionSaved() {
+    std::string newIdentifier = _actions.identifier->text().toStdString();
+    if (newIdentifier.empty()) {
+        QMessageBox::critical(this, "Empty identifier", "Identifier must not be empty");
+        return;
+    }
+
+    Profile::Action* action = selectedAction();
+    std::string oldIdentifier = action->identifier;
+    if (oldIdentifier != newIdentifier) {
+        // The identifier is a bit special as we need to make sure that we didn't
+        // accidentally create a duplicate.  Also if we did create a duplicate we need to
+        // make sure that we update all keybinds that referenced the old value are updated
+        // to the new one
+
+        const auto it = std::find_if(
```

Comment:
The comment above makes me wonder if it would be possible to get more than one duplicate action? If so then this would only find the first. Maybe this should be inside of a loop that ends only when iterator == _actions.data.end()

Reply:
I rewrote the comment to make it clear that the identifier of an action has to be unique, so there is no need for a loop here as there can only be 0 or 1 hit on the check